### PR TITLE
More profiles fixes and added backups

### PR DESCRIPTION
**Pull request by DevBefell**
Changed a few things so that it would _hopefull_ have less issues of obliviating the profiles as well as a backup every startup.

### CODE REVIEW

File: launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch [hunk 1]
```diff
@@ -118,40 +126,71 @@ index bb7d8a0..70fb5f8 100644
...
+-                    this.launcher.getLauncher().shutdownLauncher();
+-                    System.exit(0);
+-                    return false;
+-                }
++            // olauncher start - creates backups for profiles
++            File oldProfile =  new File(this.profileFile.getParentFile(), "launcher_profiles.old.json");
++            if (oldProfile.exists()){
++                oldProfile.delete();
+             }
++            Files.copy(this.profileFile.toPath(), oldProfile.toPath());
```
Comment (bigfoot547) [inline launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch:157]: could probably use `REPLACE_EXISTING` (documented [here](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#copy-java.nio.file.Path-java.nio.file.Path-java.nio.file.CopyOption...-))

**Review by bigfoot547 — CHANGES_REQUESTED** [review]
just 1 quick thing

**Review by bigfoot547 — APPROVED** [review]
looks good 👍 will test and merge when I get home

**Comment by DevBefell** [comment]
Also just a side note, I used the `replaceAll` method and it is only Java 8 onwards compatible as I believe that as now using Java 8 onwards is more secure and more maintained, so basically *I haven't tested it* it wouldn't be compatible with pre Java 8

**Comment by bigfoot547** [comment]
0bc9ed5344d5167c3eb4a4b880b26128e01b8713

**Pull request closed without merging**